Using Pass Execution causes pytest failure

From the docs:

"Skips rest of the current test ..."
https://robotframework.org/robotframework/latest/libraries/BuiltIn.html#Pass%20Execution

Under the hood it is implemented by raising an Exception: 
https://github.com/robotframework/robotframework/blob/master/src/robot/errors.py#L296

In my use case I would like to treat a test with this keyword as a Pass. 

I don't know if its the correct behavior for every case, but it would be useful if it were at least configurable.

In my test:
```
    Pass Execution If    '${v}'=='XXX'     foo
```
The result:
```
$ pytest tests/test.robot::case2
...
========== short test summary info ==========
FAILED tests/test.robot::case2 - robot.errors.PassExecution: foo
```

The Robot report displays it correctly, but I think probably pytest could too.
I hope this is in the scope of this package, thanks for the help!

--- a/pytest_robotframework/_internal/pytest/robot_file_support.py
+++ b/pytest_robotframework/_internal/pytest/robot_file_support.py
@@ -9,7 +9,7 @@ from typing import TYPE_CHECKING, Callable, cast, final
 from _pytest._code.code import ReprFileLocation, TerminalRepr
 from pytest import Config, ExceptionInfo, File, Item, MarkDecorator, Session, StashKey, mark, skip
 from robot import model
-from robot.errors import ExecutionFailed, ExecutionFailures, RobotError
+from robot.errors import ExecutionFailures, ExecutionStatus, RobotError
 from robot.libraries.BuiltIn import BuiltIn
 from robot.running.bodyrunner import BodyRunner
 from robot.running.model import Body
@@ -112,19 +112,22 @@ class RobotItem(Item):  # pyright:ignore[reportUninitializedInstanceVariable]
 
     @staticmethod
     @contextmanager
-    def _check_skipped() -> Iterator[None]:
-        """since robot and pytest skips are different, we need to catch robot skips and convert them
-        to pytest skips"""
+    def _check_execution_status() -> Iterator[None]:
+        """
+        catches robot execution status exceptions to turn them into their pytest equivalent
+        """
         try:
             yield
-        except ExecutionFailed as e:
+        except ExecutionStatus as e:
             if e.status == "SKIP":
                 skip(e.message)
-            raise
+            if e.status != "PASS":  # pyright:ignore[reportUnnecessaryComparison] type is wrong
+                # unlike robot, pytest does not raise a passed exception
+                raise
 
     def _run_keyword(self, keyword: model.Keyword | None):
         if keyword and keyword.name is not None and keyword.name.lower() != "none":
-            with self._check_skipped():
+            with self._check_execution_status():
                 BuiltIn().run_keyword(keyword.name, *keyword.args)
 
     @override
@@ -137,7 +140,7 @@ class RobotItem(Item):  # pyright:ignore[reportUninitializedInstanceVariable]
         context = execution_context()
         if not context:
             raise InternalError("failed to runtest because no execution context")
-        check_skipped = self._check_skipped()
+        check_skipped = self._check_execution_status()
         if robot_6:
             with check_skipped:
                 # pyright is only run when robot 7 is installed

--- a/tests/fixtures/test_robot/test_pass_execution.robot
+++ b/tests/fixtures/test_robot/test_pass_execution.robot
@@ -1,0 +1,3 @@
+*** Test Cases ***
+Asdf
+    Pass Execution    hi

--- a/tests/test_robot.py
+++ b/tests/test_robot.py
@@ -335,3 +335,8 @@ def test_keyword_decorator_class_library(pr: PytestRobotTester):
         output_xml(),
         f"//kw[@name='Foo' and @{'library' if robot_6 else 'owner'}='ClassLibrary']/msg[.='hi']",
     )
+
+
+def test_pass_execution(pr: PytestRobotTester):
+    pr.run_and_assert_result(passed=1)
+    pr.assert_log_file_exists()

